src/wp-admin/includes/file.php

```diff
@@ -1946,6 +1946,83 @@ function copy_dir( $from, $to, $skip_list = array() ) {
 	return true;
 }
 
+/**
+ * Moves a directory from one location to another.
+ *
+ * Recursively invalidates OPcache on success.
+ *
+ * If the renaming failed, falls back to copy_dir().
+ *
+ * Assumes that WP_Filesystem() has already been called and setup.
+ *
+ * @since 6.2.0
+ *
+ * @global WP_Filesystem_Base $wp_filesystem WordPress filesystem subclass.
+ *
+ * @param string $from      Source directory.
+ * @param string $to        Destination directory.
+ * @param bool   $overwrite Optional. Whether to overwrite the destination directory if it exists.
+ *                          Default false.
+
+ * @return true|WP_Error True on success, WP_Error on failure.
+ */
+function move_dir( $from, $to, $overwrite = false ) {
+	global $wp_filesystem;
+
+	if ( trailingslashit( strtolower( $from ) ) === trailingslashit( strtolower( $to ) ) ) {
+		return new WP_Error(
+			'source_destination_same_move_dir',
+			__( 'The source and destination are the same.' )
+		);
+	}
+
+	if ( ! $overwrite && $wp_filesystem->exists( $to ) ) {
+		return new WP_Error(
+			'destination_already_exists_move_dir',
+			sprintf(
+				/* translators: %s: The destination directory path. */
+				__( 'The destination folder, %s, already exists.' ),
+				"<code>$to</code>"
+			)
+		);
```

Comment:
```suggestion
		return new WP_Error( 'destination_already_exists_move_dir', __( 'The destination folder already exists' ), $to );
```

What about simply passing `$to` as a parameter to `WP_Error`?

Reply:
This is consistent with line 2012 below.

Reply:
Makes sense, this would also be consistent with similar strings in `copy_dir()` 👍 

These strings can all be improved in a later iteration to display the folder when updating (if it's not displayed already).

Reply:
Thanks to both! 🙂 Committed in [406055a](https://github.com/WordPress/wordpress-develop/pull/3909/commits/406055a23091b3f0fe5891ec073386ca78cd75d9).